integration.js

```diff
@@ -72,34 +78,70 @@ function IntegrationClient (baseUrl, applicationId, applicationKey, context) {
  * @param {Object} updates an object containing session properties to update
  * @see {@link http://developers.talon.one/integration-api/reference/#updateCustomerSession}
  */
 IntegrationClient.prototype.updateCustomerSession = function (sessionId, updates, callback) {
-  return this.request('PUT', '/v1/customer_sessions/' + sessionId, updates, callback)
+  return this.request('PUT', API_CUSTOMER_SESSIONS_PATH + sessionId, updates, callback)
 }
 
 /**
  * Update/create a customer profile
  *
- * @param {string} customerId The integration ID of the customer
+ * @param {string} customerId: The integration ID of the customer profile
  * @param {Object} updates an object containing profile properties to update
  * @see {@link http://developers.talon.one/integration-api/reference/#updateCustomerProfile}
  */
 IntegrationClient.prototype.updateCustomerProfile = function (customerId, updates, callback) {
-  return this.request('PUT', '/v1/customer_profiles/' + customerId, updates, callback)
+  return this.request('PUT', API_CUSTOMER_PROFILES_PATH + customerId, updates, callback)
 }
 
 /**
  * Track a custom event
  *
- * @param {string} sessionId The integration ID of the customer
- * @param {Object} updates an object containing profile properties to update
+ * @param {string} sessionId:   The integration ID of the customer session for this event
+ * @param {string} customerId:  An optional integration ID of the customer profile for this event
+ * @param {string} eventType:   A string representing the event. Must not be a reserved event name
+ * @param {Object} eventData:   Arbitrary data asspcoated wotj tjos evemt
  * @see {@link http://developers.talon.one/integration-api/reference/#trackEvent}
  */
-IntegrationClient.prototype.trackEvent = function (sessionId, eventType, eventData, callback) {
-  return this.request('POST', '/v1/events', {sessionId: sessionId, type: eventType, value: eventData}, callback)
+IntegrationClient.prototype.trackEvent = function (sessionId, customerId, eventType, eventData, callback) {
+  return this.request('POST', API_EVENTS_PATH, {sessionId: sessionId, profileId: customerId, type: eventType, attributes: eventData}, callback)
+}
+
+/**
+ * Create a referral code
+ *
+ * @param {number} campaignId:  ID of the Campaign for which the referral code is being created
+ * @param {string} advocateId:  The profile integration ID of the referral's advocate
+ * @param {string} friendId:    The optional profile integration ID of the friend being referred
+ * @param {string} start:       An optional string with an RFC3339 timestamp to set a start date for validation
+ * @param {string} end:         An optional string with an RFC3339 timestamp to set an end date for validation
+ * @see {@link http://developers.talon.one/integration-api/reference/#createReferral}
+ */
+IntegrationClient.prototype.createReferral = function (campaignId, advocateId, friendId, start, end, callback) {
```

Comment:
Would prefer that `friendId`, `start` and `end` were combined into a single `options` object.